igvc_navigation/GraphSearch/include/Graph.h

```diff
@@ -27,10 +27,68 @@ Date Created: December 16, 2018
 #include <cassert>
 #include <cmath>
 #include <limits>
+#include <queue>
 #include <tuple>
 #include <utility>
 #include <vector>
 
+#define NDEBUG
+
+/**
+A Position represents any discrete or continuous point in the Graph space. A
+Position differs from a Node in that it is not restricted to lying on a vertex, but
+can also lie along an edge. As such, a position is represented by two floats <x,y>
+that represent the position's x & y coordinates in the Graph.
+*/
+struct Position
+{
+  float x;
+  float y;
+
+  Node castToNode() const
+  {
+    return Node(static_cast<int>(this->x), static_cast<int>(this->y));
+  }
+
+  Position()
+  {
+  }
+
+  Position(float x, float y)
+  {
+    this->x = x;
+    this->y = y;
+  }
+
+  Position(std::tuple<float, float> position) : Position(std::get<0>(position), std::get<1>(position))
+  {
+  }
+
+  Position(Node n) : Position(static_cast<std::tuple<float, float>>(n.getIndex()))
+  {
+  }
+
+  // overloaded assignment operator
+  Position& operator=(const Position& other)
```

Comment:
Why do you explicitly define this, as well as `==` and `!=`? Pretty sure c++ by default implements these for you.

Reply:
I don't think this is correct. How would C++ know what fields I want to compare for equality? The default `==` operator checks for reference equality if i'm not mistaken.

Reply:
Oh woops, for some reason I was under the impression that C++ had default struct comparison for some reason.